### PR TITLE
[devex] add vite

### PR DESCRIPTION
**Pull request by guillaumejparis**
Add vite for local development

### CODE REVIEW

File: opencti-platform/opencti-front/.eslintrc.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ module.exports = {
     'plugin:@typescript-eslint/recommended',
     'plugin:react/recommended',
   ],
+  settings: {
+    react: {
+      version: 'detect',
+    },
+  },
```
Comment (guillaumejparis) [inline opencti-platform/opencti-front/.eslintrc.js:14]: Fix for the eslint plugin react warning "en passant"

File: opencti-platform/opencti-front/index.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        <meta name="dеѕсrірtіоn" content="%APP_DESCRIPTION%">
+        <link id="favicon" rel="shortcut icon" href="%APP_FAVICON%">
+        <link id="manifest" rel="manifest" href="%APP_MANIFEST%">
+    </head>
+    <body>
+        <noscript>You need to enable JavaScript to run this app.</noscript>
+        <div id="root"></div>
+        <script type="module" src="/src/front.tsx"></script>
+    </body>
+</html>
```
Comment (guillaumejparis) [inline opencti-platform/opencti-front/index.html:18]: For Vite to work we MUST have the index.html at the root

File: opencti-platform/opencti-front/package.json [hunk 3]
```diff
@@ -142,6 +143,7 @@
     "build": "yarn relay && yarn esbuild",
     "build:standalone": "yarn relay && yarn esbuild --keep",
     "start": "yarn relay && node builder/dev/dev.js",
+    "dev": "yarn relay && vite",
```
Comment (guillaumejparis) [inline opencti-platform/opencti-front/package.json:143]: At first vite is optin with the `yarn dev` command

File: opencti-platform/opencti-front/vite.config.mts [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+      async transform(code, id) {
+        if (!id.match(/src\/.*\.js$/)) return null;
+        // Use the exposed transform from vite, instead of directly
+        // transforming with esbuild
+        return transformWithEsbuild(code, id, {
+          loader: 'tsx',
+          jsx: 'automatic',
+        });
+      },
+    },
```
Comment (guillaumejparis) [inline opencti-platform/opencti-front/vite.config.mts:208]: needed because we have js files with jsx inside

**Comment by guillaumejparis** [comment]
With the addition of Vite, a little performance problem has been spotted : at the first Vite launch a lot of files are downloaded by the browser (every src/ files + dependencies chunks), to ease this some main routes are now lazy loaded, in the future we maybe will use it to improve production first render.

File: opencti-platform/opencti-front/vite.config.mts [hunk 5]
```diff
@@ -0,0 +1,226 @@
+import { createLogger, defineConfig, transformWithEsbuild } from 'vite';
+import react from '@vitejs/plugin-react';
+import * as path from 'node:path';
+import relay from 'vite-plugin-relay';
+
+// to avoid multiple reload when discovering new dependencies after a going on a lazy (not precedently) loaded route we pre optmize these dependencies
```
Comment (guillaumejparis) [inline opencti-platform/opencti-front/vite.config.mts:6]: to avoid multiple reload AND to avoid changing every mui imports across the app

**Review by lndrtrbn — APPROVED** [review]
Tested locally with both start and dev command. Looks good to me for a merge

**Comment by labo-flg** [comment]
Tested ok, good job LGTM 🙌

We'll use the remaining time in this dev cycle to stabilize or, if this happens to be really a bad idea, revert the changes.

**Review by labo-flg — APPROVED** [review]

**Pull request merged**